pkg/controller/add.go

```diff
@@ -66,5 +74,43 @@ func AddToManagerWithOptions(ctx context.Context, mgr manager.Manager, opts *Add
 		Predicates:        extension.DefaultPredicates(ctx, mgr, DefaultAddOptions.IgnoreOperationAnnotation),
 		Type:              Type,
 		ExtensionClass:    opts.ExtensionClass,
+		WatchBuilder:      watchInfrastructure(mgr),
+	})
+}
+
+func infrastructurePredicate() predicate.TypedFuncs[*extensionsv1alpha1.Infrastructure] {
+	return predicate.TypedFuncs[*extensionsv1alpha1.Infrastructure]{
+		UpdateFunc: func(e event.TypedUpdateEvent[*extensionsv1alpha1.Infrastructure]) bool {
+			// We want to reconcile if the status of the Infrastructure changed
+			return !apiequality.Semantic.DeepEqual(e.ObjectOld.Status, e.ObjectNew.Status)
```

Comment:
While testing your change I observed that the ACL extension is multiple times during `Infrastructure` reconciliation. I.e., there are several call updating `Infrastructure.status.lastOperation`, all of which trigger an `Extension` reconciliation.

This seems undesirable. Can we restrict the predicate to only react on changes to the relevant fields (`egressCIDRs`)?